### ISSUE TITLE
Dashing and jumping and then hitting a wall stops all horizontal movement, even if there is no longer contact with the wall

### ISSUE THREAD

**Issue by JosephAQM** [ldkjf2nd/17-18Capstone#32]
1. Double Tap D to dash
2. While holding D, Jump
3. Collide into a wall during the rising portion of the jump
4. Clear the wall after running into it

Once the character is above the wall, he still does not continue moving sideways (despite no longer touching the wall). This makes jumping movement feel clunky.

Hard to explain... Example: https://gfycat.com/FemaleElaborateCoral

**Comment by thatstoofar** [comment]
Yeah i'm aware of what you are talking about.  When you hit into wall when dashing which sets your horizontal velocity to 0. As you ride up the wall you only have  a vertical velocity, the left you are holding isn't adding any new forces. So if you press A or D after the initial collision you will have a horizontal force
